src/wirecloud/commons/static/js/StyledElements/ModelTable.js

```diff
@@ -195,97 +196,120 @@
+        StyledElements.StyledElement.call(this, ['click']);
 
-        /**
-         * select attribute
-         */
         var selection = [];
-        Object.defineProperty(this, 'selection', {
-            get: function () {
-                return selection;
-            },
-            set: function (value) {
-                if (!Array.isArray(value)) {
-                    throw new TypeError();
-                }
+        var source;
+        if (options.source != null) {
+            source = options.source;
+        } else if (options.pagination != null) {
+            // Backwards compatilibity
+            source = options.pagination;
+        } else {
+            sort_info = {};
+            for (i = 0; i < columns.length; i += 1) {
+                column = columns[i];
 
-                // Unhighlihgt previous selection
-                if (selection != null) {
-                    selection.forEach(function (id) {
-                        if (id in this._current_elements) {
-                            this._current_elements[id].row.classList.remove('highlight');
-                        }
-                    }, this);
+                if (sort_id in column) {
+                    sort_id = column.sort_id;
+                } else {
+                    sort_id = column.field;
                 }
-                selection = value;
+                sort_info[sort_id] = column;
+            }
+            source = new StyledElements.StaticPaginatedSource({pageSize: options.pageSize, sort_info: sort_info});
+        }
+
+        this[_attrs.layout] = new StyledElements.VerticalLayout({'class': className});
+
+        Object.defineProperties(this, {
+            columns: {
+                writable: true,
+                value: columns
+            },
+            emptyMessage: {
+                writable: true,
+                value: options.emptyMessage
+            },
+            selection: {
+                get: function () {
+                    return selection;
+                },
+                set: function (value) {
+                    if (!Array.isArray(value)) {
+                        throw new TypeError();
+                    }
+
+                    // Unhighlihgt previous selection
+                    if (selection != null) {
+                        selection.forEach(function (id) {
+                            if (id in this[_attrs.current_elements]) {
+                                this[_attrs.current_elements][id].row.classList.remove('highlight');
+                            }
+                        }, this);
+                    }
+                    selection = value;
 
-                // Highlight the new selection
-                highlight_selection.call(this);
+                    // Highlight the new selection
+                    highlight_selection.call(this);
+                }
+            },
+            source: {
+                writable: true,
```

Comment:
Use `writable: false` for now. Would be great to be able to change the source on the fly, but then, you should provide a getter for adding the needed listeners, etc...